lib/Cake/Model/Datasource/DboSource.php

```diff
@@ -382,11 +391,12 @@ public function value($data, $column = null, $null = true) {
 				if (is_float($data)) {
 					return str_replace(',', '.', strval($data));
 				}
-				if ((is_int($data) || $data === '0') || (
+				if (((is_int($data) || $data === '0') || (
 					is_numeric($data) &&
 					strpos($data, ',') === false &&
 					$data[0] != '0' &&
 					strpos($data, 'e') === false)
+					) && empty($isStringEnum)
```

Comment:
You should probably default `isStringEnum` to false/null. That would let you not use empty and check its truthyness instead.